fil_fungible_token/src/runtime/messaging.rs

```diff
@@ -47,29 +43,17 @@ pub trait Messaging {
 pub struct FvmMessenger {}
 
 impl Messaging for FvmMessenger {
-    fn call_receiver_hook(
-        &self,
-        from: ActorID,
-        to: ActorID,
-        token_value: &TokenAmount,
-        data: &[u8],
-    ) -> Result<Receipt> {
+    fn call_receiver_hook(&self, to: &Address, params: &TokenReceivedParams) -> Result<Receipt> {
         // TODO: use fvm_dispatch here (when it supports compile time method resolution)
         // TODO: ^^ necessitates determining conventional method names for receiver hooks
 
         // currently, the method number comes from taking the name as "TokensReceived" and applying
         // the transformation described in https://github.com/filecoin-project/FIPs/pull/399
         const METHOD_NUM: u64 = 1361519036;
-        let to = Address::new_id(to);
 
-        let params = TokenReceivedParams {
-            sender: Address::new_id(from),
-            value: token_value.clone(),
-            data: RawBytes::from(data.to_vec()),
-        };
         let params = RawBytes::new(fvm_ipld_encoding::to_vec(&params)?);
 
-        Ok(send::send(&to, METHOD_NUM, params, TokenAmount::zero())?)
+        Ok(send::send(to, METHOD_NUM, params, TokenAmount::zero())?)
```

Comment:
Now that we're here, maybe even METHOD_NUM should be pushed out so that this _only_ wraps the SDK method. This could then form a generic interface that would be useful for other libraries too.